judge/views/problem.py

```diff
@@ -425,8 +426,9 @@ def get_contest_queryset(self):
             'points': p['points'],
             'partial': p['partial'],
             'user_count': p['user_count'],
-        } for p in queryset.values('problem_id', 'problem__code', 'problem__name', 'i18n_name',
-                                   'problem__group__full_name', 'points', 'partial', 'user_count')]
+            'types_list': q.problem.types_list,
+        } for p, q in zip(queryset.values('problem_id', 'problem__code', 'problem__name', 'i18n_name',
+                                          'problem__group__full_name', 'points', 'partial', 'user_count'), queryset)]
```

Comment:
`zip(queryset.values(...), queryset)` is pretty weird, is there a better way to do this?

Reply:
`zip(queryset.values(...), queryset)` defeats the point of having a `queryset.values(...)`

Reply:
> `zip(queryset.values(...), queryset)` is pretty weird, is there a better way to do this?

Couldn't think of a solution using queryset.values(...) since I believe it can't access the types_list property. Maybe getting rid of the queryset.values(...) would be better?

Reply:
The `zip` is fundamentally broken.